src/datatypes/array.c

```diff
@@ -2,59 +2,55 @@
 #include "../util/arr.h"
 #include <limits.h>
 
-SIValue Array_New(u_int32_t initialCapacity) {
+SIValue SIArray_New(uint32_t initialCapacity) {
 	SIValue siarray;
 	siarray.array = array_new(SIValue, initialCapacity);
 	siarray.type = T_ARRAY;
 	siarray.allocation = M_SELF;
 	return siarray;
 }
 
-SIValue Array_Append(SIValue siarray, SIValue value) {
+void SIArray_Append(SIValue *siarray, SIValue value) {
```

Comment:
Remind me why did we changed that to a pointer?
was that to avoid copying a stack based parameter?

Reply:
yes